Lombiq.Hosting.Tenants.FeaturesGuard/Handlers/FeaturesEventHandler.cs

```diff
@@ -54,56 +51,63 @@ async Task IFeatureEventHandler.DisabledAsync(IFeatureInfo feature)
     /// Enables conditional features (key) if one of their corresponding condition features (value) was enabled.
     /// </summary>
     /// <param name="featureInfo">The feature that was just enabled.</param>
-    public async Task EnableConditionallyEnabledFeaturesAsync(IFeatureInfo featureInfo)
+    public Task EnableConditionallyEnabledFeaturesAsync(IFeatureInfo featureInfo)
     {
-        if (_shellSettings.IsDefaultShell() ||
-            _conditionallyEnabledFeaturesOptions.Value.EnableFeatureIfOtherFeatureIsEnabled is not { } conditionallyEnabledFeatures)
+        // We are using a Deferred Task so we can avoid recursive executions upon tenant setup.
+        ShellScope.AddDeferredTask(async scope =>
```

Comment:
Shouldn't `DisableConditionallyEnabledFeaturesAsync` use a deferred task as well?

Reply:
For safety measures we can wrap that with `DeferredTask` but I haven't experienced malfunctionalities without it.

Reply:
I see, then please add a comment into `DisableConditionallyEnabledFeaturesAsync` about "In my opinion it doesn't need to use DeferredTask because disabling features never happens continuously N times after each other while Enabling features do when a Tenant is being initialized."